src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts

```diff
@@ -180,12 +182,30 @@ export class DotEditContentHtmlService {
      * @param {ContentType} form
      * @memberof DotEditContentHtmlService
      */
-    renderAddedForm(form: ContentType): void {
-        this.renderAddedItem({
-            item: form,
-            checkExistFunc: this.isFormExistInContainer.bind(this),
-            getContent: this.dotContainerContentletService.getFormToContainer.bind(this.dotContainerContentletService)
-        });
+    renderAddedForm(form: ContentType): Observable<DotPageContainer[]> {
+        const doc = this.getEditPageDocument();
+        const containerEl = doc.querySelector(
+            // tslint:disable-next-line:max-line-length
+            `div[data-dot-object="container"][data-dot-identifier="${this.currentContainer.identifier}"][data-dot-uuid="${
+                this.currentContainer.uuid
+            }"]`
+        );
+
+        if (this.isFormExistInContainer(form, containerEl)) {
+            this.showContentAlreadyAddedError();
+            return Observable.of(null);
+        } else {
+            return this.dotContainerContentletService.getFormToContainer(this.currentContainer, form).map(response => {
+                const containers: DotPageContainer[]  = this.getContentModel();
+
+                containers.filter(container =>
+                    container.identifier === this.currentContainer.identifier && container.uuid === this.currentContainer.uuid)
+                    .filter(container => container.contentletsId)
```

Comment:
I think you don't need 2 `filter` here, you can add the condition of the second to the first with an `&&` and maybe because the condition is so long you might want to create a method for that.